llvmlite/tests/test_binding.py

```diff
@@ -1273,22 +1274,82 @@ def test_inlineasm(self):
 
 
 class TestObjectFile(BaseTest):
+
+    mod_asm = """
+        ;ModuleID = <string>
+        target triple = "{triple}"
+
+        declare i32 @sum(i32 %.1, i32 %.2)
+
+        define i32 @sum_twice(i32 %.1, i32 %.2) {{
+            %.3 = add i32 %.1, %.2
+            %.4 = add i32 %.3, %.3
+            ret i32 %.4
+        }}
+    """
+
     def test_object_file(self):
         target_machine = self.target_machine()
         mod = self.module()
         obj_bin = target_machine.emit_object(mod)
         obj = llvm.ObjectFileRef.from_data(obj_bin)
         # Check that we have a text section, and that she has a name and data
         has_text = False
+        last_address = -1
         for s in obj.sections():
             if s.is_text():
                 has_text = True
                 self.assertIsNotNone(s.name())
                 self.assertTrue(s.size() > 0)
                 self.assertTrue(len(s.data()) > 0)
+                self.assertIsNotNone(s.address())
+                self.assertTrue(last_address < s.address())
+                last_address = s.address()
                 break
         self.assertTrue(has_text)
 
+    def test_add_object_file(self):
+        target_machine = self.target_machine()
+        mod = self.module()
+        obj_bin = target_machine.emit_object(mod)
+        obj = llvm.ObjectFileRef.from_data(obj_bin)
+
+        jit = llvm.create_mcjit_compiler(self.module(self.mod_asm),
+            target_machine)
+
+        jit.add_object_file(obj)
+
+        sum_twice = CFUNCTYPE(c_int, c_int, c_int)(
+            jit.get_function_address("sum_twice"))
+
+        assert sum_twice(2, 3) == 10
+
+    def test_add_object_file_from_filesystem(self):
```

Comment:
For this test, I'm sometimes getting. `LLVM ERROR: Malformed MachO file` or just a bus error.

Reply:
@stuartarchibald said he cannot replicate this on Linux.

Will need to do a debug build of llvm to check this.

Reply:
On a related note: is there a way to build conda `llvmdev` recipe with debugging flags?

Reply:
This problem seems to be resolved by llvm7.

Reply:
The problem was not resolved by llvm7.  There was a ownership issue with the `LLVMPY_CreateObjectFile`.  It was more apparent in the test against pypy as the interpreter in Azure.  I fixed it in https://github.com/numba/llvmlite/pull/389/commits/da02109703211a0b0ab0798e2068483c6db002dc.